javascript/ui.js

```diff
@@ -427,12 +427,69 @@ async function browseModels(){
     }
 }
 
+function searchModel({model_type, searchValue}) {
+    return fetch(`/sd_extra_networks/update_page?model_type=${model_type}&page=1&search_value=${searchValue}&page_size=10&need_refresh=false`, {
+        method: "GET", cache: "no-cache"});
+}
+
+async function getModelFromUrl() {
+
+    // get model form url
+    const urlParam = new URLSearchParams(location.search);
+    const checkpointModelValueFromUrl = urlParam.get('checkpoint').toLowerCase();
+
+    document.cookie = `selected_checkpoint_model=${checkpointModelValueFromUrl}`;
+    const keyMapModelType = {
+        "checkpoint": "checkpoints",
+        "lora": "lora",
+        "ti": "textual_inversion",
+        "hn": "hypernetworks"
+    }
+
+    const promiseList = [];
+    const urlList = urlParam.entries();
+    const urlKeys =  [];
+    const urlValues =  [];
+
+    for (const [key, value] of urlList) {
+        if (keyMapModelType[key]) {
```

Comment:
You did not capture the error where there are multiple `checkpoint`in the url. Since there could only be a single `checkpoint` model at a time, you should at least notify user there should not be multple `checkpoint` in the url and only the first one will be used.